image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -148,38 +155,58 @@ class ImageLoaderController(auth: Authentication,
       )
 
       logger.info("importImage request start")
+      val existingImageStatus = uploadStatusTable.getStatus(uri.split("/").last).map {
+        case Some(Right(record)) => (record.uploadedBy, record.identifiers, record.fileName)
+        case _ => (None, None, None)
+      }.recover{case _ => (None, None, None)}
 
       val tempFile = createTempFile("download")
-      val result = for {
+      val importResult = for {
+        record <- existingImageStatus
         validUri <- Future { URI.create(uri) }
         digestedFile <- downloader.download(validUri, tempFile)
         uploadRequest <- uploader.loadFile(
           digestedFile,
           request.user,
-          uploadedBy,
-          identifiers,
+          record._1.orElse(uploadedBy),
```

Comment:
It's best to avoid `_N` notation as it's hard to comprehend what the content of each `N` is (more so if the assignment is several lines earlier). I think here it would be better if `record` was an option. Then you can do `maybeRecord.flatMap(_.uploadedBy).orElse(uploadedBy)`.